src/main/java/world/reports/NPopulatedCitiesWithinRegion.java

```diff
@@ -19,14 +19,16 @@
 
             String strSelect = "SELECT ci.Name, ci.Population" +
                     "FROM city ci " +
+                    "JOIN country c on ci.country = c.name" +
+                    "WHERE c.region = REGION" +
                     "ORDER BY ci.Population DESC;" +
                     "LIMIT N";
 
             List<city> result = new ArrayList<>();
 
             ResultSet rSet = stmt.executeQuery(strSelect);
```

Comment:
Now here, we will need to add our parameter to the statement: (where 1 says it's the first parameter in the query - the first question mark found, and `numberOfRows` is the value received by the function)
```suggestion
            stmt.setInteger(1, numberOfRows);
            ResultSet rSet = stmt.executeQuery();
```